js/state-from-store-mixin.jsx

```diff
@@ -140,16 +150,11 @@ var StateFromStore = function(stateDescriptors) {
                 changeListeners: {}
             };
 
-            var initialState = {};
-            Object.keys(stateDescriptors).forEach(function(stateKey) {
-                var stateData = fetchNewStateData(
-                    this, stateKey, false).stateData;
-                initialState[stateKey] = stateData;
-            }, this);
-            return initialState;
+            return getStoreState(this);
         },
 
         componentDidMount: function() {
+            this.setState(getStoreState(this));
```

Comment:
You can call setState in componentWillMount, and it will use that state in the initial render: https://facebook.github.io/react/docs/component-specs.html#mounting-componentwillmount